libs/designer/src/lib/ui/mcp/connectors/ListConnectors.tsx

```diff
@@ -0,0 +1,233 @@
+    connectionReferences: state.connection.connectionReferences,
+  }));
+
+  const INTL_TEXT = {
+    noConnectors: intl.formatMessage({
+      id: 'xyhnsP',
+      defaultMessage: 'No connectors added yet',
+      description: 'Message displayed when no connectors are available',
+    }),
+    addFirstConnector: intl.formatMessage({
+      id: 'i/0DrA',
+      defaultMessage: 'Add your first connector to get started',
+      description: 'Message prompting the user to add their first connector',
+    }),
+    connectorLabel: intl.formatMessage({
+      defaultMessage: 'Name',
+      id: 'T1q9LE',
+      description: 'The label for the connector column',
+    }),
+    connectionLabel: intl.formatMessage({
+      defaultMessage: 'Connection',
+      id: 'cjWC0X',
+      description: 'The label for the connection column',
+    }),
+    statusLabel: intl.formatMessage({
+      defaultMessage: 'Status',
+      id: 'ozFnEE',
+      description: 'The label for the status column',
+    }),
+    tableAriaLabel: intl.formatMessage({
+      defaultMessage: 'List of connectors with their connections',
+      id: 'd9Ooue',
+      description: 'The aria label for the connections table',
+    }),
+    editButtonLabel: intl.formatMessage({
+      defaultMessage: 'Edit connector',
+      id: 'RTfra/',
+      description: 'Label for the edit connector button',
+    }),
+    deleteButtonLabel: intl.formatMessage({
+      defaultMessage: 'Delete connector',
+      id: '8e1bKU',
+      description: 'Label for the delete connector button',
+    }),
+    connectedText: intl.formatMessage({
+      defaultMessage: 'Connected',
+      id: 'V7NT3q',
+      description: 'Text indicating a connector is connected',
+    }),
+    disconnectedText: intl.formatMessage({
+      defaultMessage: 'Disconnected',
+      id: 'ssR+UG',
+      description: 'Text indicating a connector is disconnected',
+    }),
+  };
+
+  const styles = useConnectorSectionStyles();
+
+  const items = useMemo(() => {
+    const seen = new Set<string>();
+
+    return Object.values(operationInfos).reduce<
+      Array<{
+        connectorId: string;
+        displayName?: string;
+        iconUri?: string;
+        connectionName: string;
+        isConnected: boolean;
+      }>
+    >((acc, info) => {
+      const connectorId = info?.connectorId;
+      if (!connectorId || seen.has(connectorId)) {
+        return acc;
+      }
+
+      seen.add(connectorId);
+
+      const metadata = operationMetadata[info.operationId];
+      const referenceKey = connectionsMapping[info.operationId];
+      const reference = referenceKey ? connectionReferences[referenceKey] : null;
+
+      const isConnected = !!reference;
+      const connectionName = isConnected
+        ? (reference.connectionName ?? getResourceNameFromId(reference.connection?.id) ?? 'Default Connection')
+        : referenceKey === null
+          ? 'No Connection'
+          : 'Default Connection';
```

Comment:
We need connection display name from connection object, useConnection there is some query cache